core/src/test/java/org/bitcoinj/core/Base58DecodeTest.java

```diff
@@ -0,0 +1,50 @@
+package org.bitcoinj.core;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class Base58DecodeTest {
+
+    private String input;
+    private byte[] expected;
+
+    public Base58DecodeTest(String input, byte[] expected) {
+        this.input = input;
+        this.expected = expected;
+    }
+
+    @Parameters
+    public static Collection<Object[]> parameters() {
+        return Arrays.asList(new Object[][]{
+                {"JxF12TrwUP45BMd", "Hello World".getBytes()},
+                {"1", new byte[1]},
+                {"1111", new byte[4]}
+        });
+    }
+
+    @Test
+    public void testDecode() {
+        byte[] actualBytes = Base58.decode(input);
+        assertArrayEquals(input,  actualBytes, expected);
+    }
+
+    @Test
```

Comment:
This test (and the next) run 3 times though it doesn't use the different input data.

Reply:
True. Do you want me to add two seperate test classes for those or refactor the test using a theory, then I can include an assumption about the string being valid base58 or an empty string.

Reply:
If there is an easy way to keep the test in the same class then I think I'd prefer it that way.

Reply:
I don't think there is with Parametrized tests. The problem with using Theories is that I cannot easily include the expected values to compare to the inputs used.

Reply:
Ok, then let's keep it as is.